utils/results/conversion/simplejsonparser/simplejsonparser.go

```diff
@@ -65,7 +65,35 @@ func (sjc *CmdResultsSimpleJsonConverter) ParseNewTargetResults(target results.S
 	return
 }
 
-func (sjc *CmdResultsSimpleJsonConverter) ParseViolations(target results.ScanTarget, scaResponse services.ScanResponse, applicabilityRuns ...*sarif.Run) (err error) {
+// We only care to update the status if it's the first time we see it or if status is 0 (completed) and the new status is not (failed)
+func shouldUpdateStatus(currentStatus, newStatus *int) bool {
+	if currentStatus == nil || (*currentStatus == 0 && newStatus != nil) {
+		return true
+	}
+	return false
+}
+
+func (sjc *CmdResultsSimpleJsonConverter) ParseScaIssues(target results.ScanTarget, violations bool, scaResponse results.ScanResult[services.ScanResponse], applicableScan ...results.ScanResult[[]*sarif.Run]) (err error) {
+	if sjc.current == nil {
+		return results.ErrResetConvertor
+	}
+	if shouldUpdateStatus(sjc.current.Statuses.ScaStatusCode, &scaResponse.StatusCode) {
+		sjc.current.Statuses.ScaStatusCode = &scaResponse.StatusCode
+	}
+	for i := range applicableScan {
+		if shouldUpdateStatus(sjc.current.Statuses.ApplicabilityStatusCode, &applicableScan[i].StatusCode) {
+			sjc.current.Statuses.ApplicabilityStatusCode = &applicableScan[i].StatusCode
+		}
+	}
+	if violations {
```

Comment:
can we have both violations and vulnerabilities (includeAllVulnerabilities)

Reply:
yes we can, but we parse them at separate time, it will call `ParseScaIssues` two times if needed, one time for vulnerabilities and one time for violations. the logic to decide if we include vuln, vio or both is in `convertor.go` so it will apply to all formats.